chain/network/src/peer_manager.rs

```diff
@@ -828,8 +829,21 @@ impl Handler<NetworkRequests> for PeerManagerActor {
                 // Filter known accounts before validating them.
                 let new_accounts = accounts
                     .into_iter()
-                    .filter(|announce_account| {
-                        !self.routing_table.contains_account(&announce_account)
+                    .filter_map(|announce_account| {
+                        if let Some(current_announce_account) =
```

Comment:
How does this code solve the problem? It seems that sending the same account with different peer id and epoch id will still be considered a new announcement? So if someone changes their peer id for the next epoch wouldn't the same bug be triggered?

Reply:
Note that currently we only maintain all the time at most one `PeerId` per `AccountId`. I make a simple filter at this moment which is, filter out  announcement that contains the same epoch that we currently have. But remaining announcements are sent to the client, filtered there (comparing if new `EpochId` is actually newer), and sent back to `PeerManager` those which passed filter.